Sources/TuistCore/Models/Scheme.swift

```diff
@@ -85,16 +85,30 @@ public class ExecutionAction: Equatable {
     }
 }
 
+public struct TargetReference: Equatable {
```

Comment:
Shouldn't this model have an equivalent in the `ProjectDescription` framework? Otherwise we are breaking the parity between the 2 and just adding features for the users of the `TuistGenerator`.

Reply:
As part of https://github.com/tuist/tuist/issues/667 the idea was to first change the `SchemesGenerator` methods without affecting the project scheme customisation API for `ProjectDescription` users i.e., adding no new features. 

I was thinking in follow up PRs we can add support for customising workspace schemes and then enhancing the project scheme API to support referencing targets in other projects. Both of these features will be exposed via the `ProjectDescription`. 

It was broken up this way mostly to reduce the size of the PRs.

Reply:
> I was thinking in follow up PRs we can add support for customising workspace schemes and then enhancing the project scheme API to support referencing targets in other projects. Both of these features will be exposed via the ProjectDescription.

Awesome! That's a good plan